pyanaconda/modules/subscription/subscription.py

```diff
@@ -17,24 +17,58 @@
 # License and may only be used or replicated with the express permission of
 # Red Hat, Inc.
 #
+import pydbus
+
+from pyanaconda.core import util
+
 from pyanaconda.dbus import DBus
+from pyanaconda.dbus.typing import get_variant, Str
+from pyanaconda.dbus.connection import Connection
 from pyanaconda.core.signal import Signal
 from pyanaconda.modules.common.base import KickstartModule
 from pyanaconda.modules.common.constants.services import SUBSCRIPTION
+from pyanaconda.modules.common.constants.services import RHSM
+from pyanaconda.modules.common.constants.objects import RHSM_CONFIG, RHSM_REGISTER_SERVER
 from pyanaconda.modules.subscription.subscription_interface import SubscriptionInterface
 from pyanaconda.modules.subscription.kickstart import SubscriptionKickstartSpecification
+from pyanaconda.modules.subscription.constants import AuthenticationMethod
 from pyanaconda.modules.subscription.installation import SystemPurposeConfigurationTask
+from pyanaconda.modules.subscription.installation import StartRHSMTask, RegisterWithUsernamePasswordTask
+from pyanaconda.modules.subscription.installation import RegisterWithOrganizationKeyTask, AttachSubscriptionTask
+from pyanaconda.modules.subscription.installation import UnregisterTask
+from pyanaconda.modules.subscription.installation import TransferSubscriptionTokensTask
+from pyanaconda.modules.subscription.installation import ConnectToInsightsTask
 from pyanaconda.modules.subscription import system_purpose
 
 from pyanaconda.anaconda_loggers import get_module_logger
 log = get_module_logger(__name__)
 
 
+# FIXME: only leave this in installation.py ?
+class RHSMPrivateBusConnection(Connection):
+    """Representation of a RHSM private bus connection."""
+
+    def __init__(self, rhsm_private_bus_address):
+        super().__init__()
+        self._rhsm_private_bus_address = rhsm_private_bus_address
+
+    def get_new_connection(self):
+        """Connect to the RHSM private bus."""
+        log.info("Connecting to RHSM private bus at %s.", self._rhsm_private_bus_address)
+        connection = pydbus.bus.Gio.DBusConnection.new_for_address_sync(self._rhsm_private_bus_address,
```

Comment:
I think we have a better solution than using `pydbus` directly. @poncovka could you please look on this?